src/parachain/issue.ts

```diff
@@ -121,33 +159,73 @@ export class DefaultIssueAPI extends DefaultTransactionAPI implements IssueAPI
+                maxTotal = maxTotal.plus(vaultAvailable);
+                singleVault = singleVault.gt(vaultAvailable) ? singleVault : vaultAvailable;
+                return [singleVault, maxTotal];
+            },
+            [new Big(0), new Big(0)]
+        );
+        return { singleVaultMaxIssuable, totalMaxIssuable };
+    }
+
     /**
      * @param events The EventRecord array returned after sending an issue request transaction
      * @returns The issueId associated with the request. If the EventRecord array does not
      * contain issue request events, the function throws an error.
      */
-    private getRequestIdFromEvents(events: EventRecord[]): Hash {
-        for (const { event } of events) {
-            if (this.api.events.issue.RequestIssue.is(event)) {
-                const hash = this.api.createType("Hash", event.data[0]);
-                return hash;
+    private getIssueIdsFromEvents(events: EventRecord[]): Hash[] {
+        return getRequestIdsFromEvents(events, this.api.events.issue.RequestIssue, this.api);
+    }
+
+    async request(
+        amount: Big,
+        options?: {
+            availableVaults?: Map<AccountId, Big>,
+            atomic?: boolean,
+            retries?: number,
+        }
+    ): Promise<IssueRequestResult[]> {
+        try {
+            const availableVaults = options?.availableVaults || await this.vaultsAPI.getVaultsWithIssuableTokens();
+            const atomic = !!options?.atomic;
+            const retries = options?.retries || 0;
+            const amountsPerVault = allocateAmountsToVaults(availableVaults, amount);
+            const result = await this.requestAdvanced(amountsPerVault, atomic);
+            const successfulSum = result.reduce((sum, req) => sum.plus(req.issueRequest.amount.toString()), new Big(0));
+            const remainder = amount.sub(successfulSum);
+            if (remainder.eq(0) || retries === 0) return result;
+            else {
+                return (await this.request(remainder, { availableVaults, atomic, retries: retries - 1 })).concat(result);
             }
+        } catch (e) {
+            return Promise.reject(e.message);
         }
-        throw new Error("Request transaction failed");
     }
 
-    async request(amount: Big, vaultId?: AccountId): Promise<IssueRequestResult> {
-        if (!vaultId) {
-            vaultId = await this.vaultsAPI.selectRandomVaultIssue(amount);
+    async requestAdvanced(
+        amountsPerVault: Map<AccountId, Big>,
+        atomic: boolean
+    ): Promise<IssueRequestResult[]> {
+        const txs = new Array<SubmittableExtrinsic<"promise">>();
+        for (const [vault, amount] of amountsPerVault) {
+            const griefingCollateral = await this.getGriefingCollateral(amount);
+            // mul() here is a hacky workaround for rounding errors
+            const griefingCollateralPlanck = new Big(dotToPlanck(griefingCollateral.toString()) || "0").mul(1.01);
```

Comment:
What is happening here? It's basically off by a couple of planck or why increase this by 1%?

Reply:
Yep, you can check the discussion in discord - the request fails with `InsufficientCollateral` without this, despite best attempts to round the plank amount upwards.

Reply:
To add, I'm testing whether this can be lowered. If it works when I add just 1-2 Planck rather than a percentage, I'll do that instead.